app/controllers/DevicesController.scala

```diff
@@ -1,41 +1,19 @@
 package controllers
 
 import model.Sensor._
-import model.{Device, Position, Sensor}
 import play.api.libs.json._
 import play.api.mvc._
+import services.DataSource
 
 
-class DevicesController(cc: ControllerComponents) extends AbstractController(cc) {
+class DevicesController(cc: ControllerComponents, ds: DataSource) extends AbstractController(cc) {
 
-  def getDevicesJson = Action {
-    Ok(Json.toJson(getDevices)).as(JSON)
-  }
-
-  def getDevicesHtml = Action {
-    Ok(views.html.devices(getDevices))
-  }
+  def getDevices(extension: String) = Action {
```

Comment:
To be honest I like it more the other way for 2 reasons:
1- The mapping is all on the routes and you don't need to look inside to see what do we accept
2- This way the complexity of the code is higher as we need to match on the extension and decide what to do with the extensions we don't handle, ie '.xml' should we respond with a 404 or a 415 or just chuck the html output and be done?